src/sentry/utils/appleconnect/appstore_connect.py

```diff
@@ -369,18 +369,30 @@ def _get_dsym_url(bundles: Optional[List[JSONData]]) -> Union[NoDsymUrl, str]:
     # was a bitcode upload or a native upload.  And whether dSYMs are
     # available for download only depends on whether it was a bitcode
     # upload.
-    if bundles is None or len(bundles) == 0:
+
+    if bundles is None:
+        bundles = []
+
+    # Remove all bundles associated with app clips, those don't have dSYMS or really any useful
+    # data since they're not apps themselves
+    app_bundles = [
+        app_bundle
+        for app_bundle in bundles
+        if safe.get_path(app_bundle, "attributes", "bundleType", default="APP") != "APP_CLIP"
+    ]
+
+    if len(bundles) == 0:
```

Comment:
```suggestion
    if len(app_bundles) == 0:
```

Reply:
You folks really need to get used to the python notation of
```suggestion
    if not app_bundles:
```

In python using `len()` is not more explicit in this case.